src/cat.jl

```diff
@@ -4,18 +4,39 @@ hcat(A::AbstractLinearOperator, B::AbstractMatrix) = hcat(A, LinearOperator(B))
 
 hcat(A::AbstractMatrix, B::AbstractLinearOperator) = hcat(LinearOperator(A), B)
 
+function hcat_prod!(res::AbstractVector, A::AbstractLinearOperator{T}, B::AbstractLinearOperator{T}, 
+                    Ancol::I, nV::I, v::AbstractVector, α, β) where {T,I<:Integer}
+  mul!(res, A, view(v, 1:Ancol), α, β)
+  mul!(res, B, view(v, (Ancol+1): nV), α, one(T))
+end
+
+function hcat_tprod!(res::AbstractVector, A::AbstractLinearOperator{T}, B::AbstractLinearOperator{T},
```

Comment:
It seems there isn’t a unit test for this? Would you mind adding one?

Reply:
Yes of course, I will add more tests later, I just needed to start the PR here to view the code coverage and see what could be improved